From 29df4771f928f10a11c3cbbd7fd41bd0d515987b Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 13 Jul 2017 20:18:08 +0100
Subject: [PATCH 08/10] sources: shutil is mortally crippled, use cp -Ra

This ensures we deduplicate where necessary, i.e. preserve the hardlinks
when "extracting" a git source. This is needed because our git build of
Papirus ballooned from 40MB to over 1GB installed.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 ypkg2/sources.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ypkg2/sources.py b/ypkg2/sources.py
index c49335a..53ee0dc 100644
--- a/ypkg2/sources.py
+++ b/ypkg2/sources.py
@@ -181,7 +181,8 @@ class GitSource(YpkgSource):
                                       "directory: {}".format(e))
                 return False
         try:
-            shutil.copytree(source, target)
+            cmd = "cp -Ra \"{}/\" \"{}\"".format(source, target)
+            subprocess.check_call(cmd, shell=True)
         except Exception as e:
             console_ui.emit_error("Git", "Failed to copy source to build")
             print(e)
-- 
2.13.3

